cmd/clusterctl/pkg/client/repository/client.go

```diff
@@ -163,8 +163,14 @@ func repositoryFactory(providerConfig config.Provider, configVariablesClient con
 		return repo, err
 	}
 
-	// if the url is a local repository
-	//TODO: implement in a follow up PR
+	// if the url is a local filesystem repository
+	if rURL.Scheme == "file" {
```

Comment:
What about adding `|| rURL.Scheme == ""` so the user can  express the URL also in a simplified way

```
url: /home/awander/.cluster-api/infra-foo/v0.1.0/infrastructure-components.yaml
```

Reply:
Done. 
Example:
```
- name: infra-foo-2
  url: /home/awander/.cluster-api/infra-foo-2/v0.2.2/infrastructure-components.yaml
  type: InfrastructureProvider
```